File: src/simpletestclient.c

//from https://www.geeksforgeeks.org/simple-client-server-application-in-c/

#include <stdio.h>
#include <errno.h>
#include <sys/socket.h>
#include <unistd.h>
#include <stdlib.h>
#include <netinet/in.h>
#include <string.h>
#include <ctype.h>
#include <netinet/tcp.h>
#include <fcntl.h>
#include <sys/types.h>

int main(int argc, char const* argv[]) 
{ 
  
    // create server socket similar to what was done in 
    // client program 
    int servSockD = socket(AF_INET, SOCK_STREAM, 0);         
    //struct linger sl;
    //sl.l_onoff = 1;
    //sl.l_linger = 0;  
    //setsockopt(servSockD,SOL_SOCKET,SO_LINGER,&sl,sizeof(sl));

    // string store data to send to client 
    char serMsg[255] = "Message from the server to the "
                       "client \'Hello Client\' "; 
  
    // define server address 
    struct sockaddr_in servAddr; 
  
    servAddr.sin_family = AF_INET; 
    servAddr.sin_port = htons(8843); 
    servAddr.sin_addr.s_addr = INADDR_ANY; 
  
    // bind socket to the specified IP and port 
    bind(servSockD, (struct sockaddr*)&servAddr, 
         sizeof(servAddr)); 
  
    // listen for connections 
    listen(servSockD, 1); 
  
    // integer to hold client socket. 
    int clientSocket = accept(servSockD, NULL, NULL); 
  
    char strData[255];
    int recstatus;
    if ((recstatus = recv(clientSocket,strData,sizeof(strData),0))<0)
    {
        printf("Failed\n");
	close(clientSocket);
	return 0;
    }
    //while (recstatus == 0)
    //{
	//recstatus = recv(clientSocket,strData,sizeof(strData),0);
    //}
    printf("Received message size %d: %s\n",recstatus,strData);
  
    //close(clientSocket);
    return 0; 
}

File: src/simpletestclient.py
import socket
import numpy as np




#create socket object
print("creating socket...",end='')
servSockD = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
print("Done!")

#bind to port number
port = 8843
print("binding to port " + str(port) + "...",end='')
servSockD.bind(('', port))
print("Done!")

#listen for conections
print("listening for connections...",end='')
servSockD.listen(1)
print("Made connection")

totalbytes = 0
while True:
    print("accepting connection...",end='')
    clientSocket,address = servSockD.accept()
    recstatus = 1
    fullMsg = ""
    print("Done!")
    print("Receiving data...")
    while recstatus > 0:
        (strData, ancdata, msg_flags, address) = clientSocket.recvmsg(255)
        recstatus = len(strData)

        #print(strData.hex(),end='')
        fullMsg += strData.hex()
        totalbytes += recstatus
    print("Done! Total bytes read:",totalbytes)

    #convert to numpy array
    print(bytes.fromhex(fullMsg)[:128])
    fullMsgBytes = bytes.fromhex(fullMsg)
    #print(fullMsgBytes[7:256+7].decode('utf-8'))
    print(np.frombuffer(fullMsgBytes[128:]).reshape(32,32,25,16))
    #print(fullMsgBytes[:])
    #print(fullMsgBytes[:].decode('utf-8'))

    clientSocket.close()

    break

print("done",totalbytes)

